src/ota_proxy/ota_cache.py

```diff
@@ -306,23 +305,31 @@ async def _reserve_space(self, size: int) -> bool:
         Returns:
             A bool indicates whether the space reserving is successful or not.
         """
-        _hashes = await self._lru_helper.rotate_cache(size)
+        _hashes = self._lru_helper.rotate_cache(size)
         # NOTE: distinguish between [] and None! The fore one means we don't need
         #       cache rotation for saving the cache file, the latter one means
         #       cache rotation failed.
         if _hashes is not None:
             logger.debug(
                 f"rotate on bucket({size=}), num of entries to be cleaned {len(_hashes)=}"
             )
-            await anyio.to_thread.run_sync(self._cache_entries_cleanup, _hashes)
+            for entry_hash in _hashes:
+                # remove cache entry
+                f = self._base_dir_sync / entry_hash
+                f.unlink(missing_ok=True)
             return True
         else:
             logger.debug(f"rotate on bucket({size=}) failed, no enough entries")
             return False
 
-    async def _commit_cache_callback(self, meta: CacheMeta):
+    def _commit_cache_callback(self, meta: CacheMeta):
```

Comment:
`commit_cache_callback` now is a sync method, executed in cache write worker thread pool.